pytket/binders/passes.cpp

```diff
@@ -145,6 +145,31 @@ const PassPtr &DecomposeClassicalExp() {
   return pp;
 }
 
+std::optional<OpTypeSet> get_gate_set(const BasePass &base_pass) {
+  OpTypeSet allowed_ops;
+  for (const std::pair<const std::type_index, std::shared_ptr<tket::Predicate>>
+           &p : base_pass.get_conditions().first) {
+    if (p.second->to_string().substr(0, 17) == "GateSetPredicate:") {
+      std::shared_ptr<GateSetPredicate> gsp_ptr =
+          std::dynamic_pointer_cast<GateSetPredicate>(p.second);
+      if (allowed_ops.empty()) {
```

Comment:
This will end up giving the wrong result in the (admittedly unlikely) case that we have already encountered two `GateSetPredicate`s with empty intersection. I suggest using a `std::optional<OpTypeSet>`, where no-value means all all ops are allowed.

Reply:
done (I think)